Update error reporting guide (#5267)
flutter/website: src/docs/cookbook/maintenance/error-reporting.md

the following steps:

  1. Get a DSN from Sentry.

>   2. Import the Sentry Flutter package.
>   3. Initialize the SDK
>   4. Capture errors programatically.

Reviewer: Also, please remove the period from item 1.

```suggestion
  2. Import the Flutter Sentry package
  3. Initialize the Sentry SDK
  4. Capture errors programmatically
```

  1. [Create an account with Sentry][].
  2. Log in to the account.

>   3. Create a new Flutter project.

Author: There's a new project type Flutter:

![image](https://user-images.githubusercontent.com/1633368/107383547-19dace80-6abf-11eb-855f-2ef4a9dd572b.png)

>   4. Copy the code snippet which includes the DSN.

Author: The installation step has a snippet which includes the DSN for the actual project:

![image](https://user-images.githubusercontent.com/1633368/107383707-4393f580-6abf-11eb-91f2-71f296babf5e.png)

>   4. Copy the code snippet which includes the DSN.

Reviewer: ```suggestion
  4. Copy the code snippet that includes the DSN.
```

>   4. Copy the code snippet which includes the DSN.

Reviewer: ```suggestion
  4. Copy the code snippet that includes the DSN.
```

## 2. Import the Sentry package

> Import the [`sentry_flutter`][] package into the app.

Author: [The `sentry` package is for pure Dart apps and the new `sentry_flutter`](https://pub.dev/packages?q=sentry) takes dependency on Flutter and integrates with it, adding native support:

![image](https://user-images.githubusercontent.com/1633368/107383851-66bea500-6abf-11eb-9111-4cf152df9e7a.png)

The sentry package makes it easier to send
error reports to the Sentry error tracking service.

```yaml
dependencies:
  sentry_flutter: <latest_version>
```

> ## 3. Initialize the SDK

Reviewer: ```suggestion
## 3. Initialize the Sentry SDK
```

> Initialize the SDK to capture different unhandled errors automatically.

Reviewer: ```suggestion
Initialize the SDK to capture different unhandled errors automatically:
```

<!-- skip -->
```dart
import 'package:flutter/widgets.dart';
import 'package:sentry_flutter/sentry_flutter.dart';

Future<void> main() async {
  await SentryFlutter.init(
    (options) => options.dsn = 'https://example@sentry.io/example',
    appRunner: () => runApp(MyApp()),
  );
}
```

> Alternatively you can pass the DSN to Flutter via Dart define:

Reviewer: ```suggestion
Alternatively, you can pass the DSN to Flutter using Dart `define`:
```

<!-- skip -->
```sh
--dart-define SENTRY_DSN=https://example@sentry.io/example
```

### What does that give me?

This is all you need for Sentry to capture unhandled errors in Dart and native layers.  

> Including Swift, Objective-C, C, C++ on iOS. And on Android: Java, Kotlin and C.

Reviewer 2: ```suggestion
Including Swift, Objective-C, C, C++ on iOS. And on Android: Java, Kotlin, C and C++.
```

Author: The thing is that it would support Rust, and other native languages, I'm not sure how much is "good enough" to list here.

> Including Swift, Objective-C, C, C++ on iOS. And on Android: Java, Kotlin, C and C++.

Reviewer: ```suggestion
This includes Swift, Objective-C, C, and C++ on iOS, and on Android: Java, Kotlin, C, and C++.
```

## 4. Capture errors programatically

> Besides the automatic error reporting that Sentry happens by initializing the SDK, you can use the API to 
> report errors to Sentry.

Reviewer: ```suggestion
Besides the automatic error reporting that Sentry generates by initializing the SDK,
you can use the API to report errors to Sentry.
```

<!-- skip -->
```dart
await Sentry.captureException(exception, stackTrace: stackTrace);
```

> The complete API is available on [pub.dev][]

Reviewer: ```suggestion
The complete API is available on [pub.dev][].
```

## Learn more

> Extensive documentation about the SDK can be found on [Sentry's documentation].

Reviewer: ```suggestion
Extensive documentation about the Sentry SDK can be found on
the [Sentry's site][].
```

[Crashy]: {{site.github}}/flutter/crashy
[Create an account with Sentry]: https://sentry.io/signup/
[Rollbar]: https://rollbar.com/
[Sentry]: https://sentry.io/welcome/
[`sentry_flutter`]: {{site.pub-pkg}}/sentry_flutter
[pub.dev]: {{site.pub-pkg}}/documentation/sentry_flutter/latest/sentry_flutter/sentry_flutter-library.html

> [Sentry's documentation]: https://docs.sentry.io/platforms/flutter/

Reviewer: ```suggestion
[Sentry's site]: https://docs.sentry.io/platforms/flutter/
```

[Sentry]: https://sentry.io/welcome/
[`sentry`]: {{site.pub-pkg}}/sentry

> [`Zone`]: {{site.api}}/flutter/dart-async/Zone-class.html

Reviewer 2: I think this could be removed as the SDK hooks up automatically

Author: Good point these are links and many of them are no longer needed

> [pub.dev]: https://pub.dev/documentation/sentry_flutter/latest/sentry_flutter/sentry_flutter-library.html

Reviewer 2: ```suggestion
[pub.dev]: {{site.pub-pkg}}/documentation/sentry_flutter/latest/sentry_flutter/sentry_flutter-library.html
```

not sure about this though
